src/Api/PersistentImplementation.hs

```diff
@@ -0,0 +1,47 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving  #-}
+{-# LANGUAGE ImpredicativeTypes          #-}
+{-# LANGUAGE OverloadedStrings           #-}
+{-# LANGUAGE ScopedTypeVariables         #-}
+{-# LANGUAGE TypeOperators               #-}
+
+{-# OPTIONS_GHC -Wall -Werror -fno-warn-orphans #-}
+
+module Api.PersistentImplementation
```

Comment:
Sorry for the short question. I would say, as we have a `Persistent` module already we could move the implementation to the `Persistent.Implementations.STM` or `Persistent.STM`. And expose the current implementation in the `Persistent` module? What do you think?